advisor/src/main/kotlin/advisors/VulnerableCode.kt

```diff
@@ -70,7 +71,10 @@ class VulnerableCode(name: String, config: AdvisorConfiguration) : Advisor(name,
                 ?: throw IOException("Unexpected URL of a vulnerability: $url.")
     }
 
-    private val vulnerableCodeConfiguration = config as VulnerableCodeConfiguration
+    private val vulnerableCodeConfiguration = config.vulnerableCode
```

Comment:
Analogously to `NexusIq`, I would propose passing the concrete `VulnerableCodeConfiguration` object from the factory as argument to the constructor and drop the `vulnerableCodeConfiguration` property.

Reply:
See the above comment.